tools/clang/lib/SPIRV/EmitVisitor.cpp

```diff
@@ -521,14 +521,21 @@ bool EmitVisitor::visit(SpirvSource *inst) {
       }
     }
 
-    // Note: in order to improve performance and avoid multiple copies, we
-    // encode this (potentially large) string directly into the debugFileBinary.
-    const auto &words = string::encodeSPIRVString(firstSnippet.getValue());
-    const auto numWordsInInstr = curInst.size() + words.size();
-    curInst[0] |= static_cast<uint32_t>(numWordsInInstr) << 16;
-    debugFileBinary.insert(debugFileBinary.end(), curInst.begin(),
-                           curInst.end());
-    debugFileBinary.insert(debugFileBinary.end(), words.begin(), words.end());
+    if (firstSnippet.hasValue()) {
+      // Note: in order to improve performance and avoid multiple copies, we
+      // encode this (potentially large) string directly into the
+      // debugFileBinary.
+      const auto &words = string::encodeSPIRVString(firstSnippet.getValue());
+      const auto numWordsInInstr = curInst.size() + words.size();
+      curInst[0] |= static_cast<uint32_t>(numWordsInInstr) << 16;
+      debugFileBinary.insert(debugFileBinary.end(), curInst.begin(),
+                             curInst.end());
+      debugFileBinary.insert(debugFileBinary.end(), words.begin(), words.end());
+    } else {
+      curInst[0] |= static_cast<uint32_t>(curInst.size()) << 16;
+      debugFileBinary.insert(debugFileBinary.end(), curInst.begin(),
+                             curInst.end());
```

Comment:
Just for my own education, do we really only realize if the file is missing when we attempt to read it?  If that's the case, should we factor out the `else` branch here with the `else` branch at line `539`?

Reply:
Yeah, this `else` clause and the one on line 539 are the same and should be factored. Probably need to change line 515 to something like
```cpp
  if (spvOptions.debugInfoSource && inst->hasFile() && firstSnippet.hasValue()) {
```